mobile_payment/gql_mutations.py

```diff
@@ -0,0 +1,421 @@
+            return InitiateTransactionMutation(uuids=transaction.uuid, internal_id=response.internal_id)
+        except Exception as exc:
+            logger.exception("transaction.mutation.failed_to_create_transaction")
+            return [{
+                    'message': _("transaction.mutation.failed_to_create_transaction"),
+                    'detail': str(exc)}] 
+        
+    
+        
+class  ProcessTransactionMutation(OpenIMISMutation):
+    """
+    create a mutaion for making a trasaction
+    """
+
+    _mutation_module = "mobile_payment"
+    _mutation_class = "ProcessTransactionMutation"
+
+    class Input(TransactionBase, OpenIMISMutation.Input):
+        pass
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **data) -> Optional[str]:
+        response = super().mutate_and_get_payload(root, info, **data)
+        try:
+            if info.context.user is AnonymousUser:
+                raise ValidationError(_("mutation.authentication_required"))
+            if not info.context.user.has_perms(MobilePaymentConfig.gql_mutation_create_transaction_perms):
+                raise PermissionDenied(_("unathorized"))
+            client_mutation_id = data.get("client_mutation_id")
+            transaction_uuid = data['uuid']
+            transacton = Transactions.objects.get(uuid=transaction_uuid )
+            if not transacton:
+                raise PermissionDenied (_("invalid transaction_uuid"))
+            transaction_id = transacton.transaction_id
+            otp = data['otp']
+            print(f' Transaction_id = {transaction_id}')
+            print (f'otp =  {otp}')
+            Qcell_process = process_request(otp, transaction_id)
+            if Qcell_process['responseCode'] == '1':
+                data['status'] = True
+                update_or_create_transaction(data, info)
+            else:
+                raise Exception(_("Failed to initiate payment"))
+            return ProcessTransactionMutation(internal_id=response.internal_id)
+        except Exception as exc:
+            logger.exception("transaction.mutation.failed_to_process_transaction")
+            return [{
+                    'message': _("transaction.mutation.failed_to_process_transaction"),
+                    'detail': str(exc)}]
+
+# *********************************** Afri Money ***********************************           
+class ProductInput(graphene.InputObjectType):
+    product_name = graphene.String(required=True)
+    product_code = graphene.String(required=True)
+    token = graphene.String(required = True)
+class ProcessPaymentInput(graphene.InputObjectType):
+    chf_id = graphene.String(required=True)
+    amount = graphene.Decimal(required=True)
+    transaction_id = graphene.String(required = False)
+    psp_service_provider_uuid = graphene.String(required = False)
+    policy = graphene.Field( ProductInput, required = False)
+    json_content = graphene.JSONString()
+
+def handle_payment(user, data):
```

Comment:
How those data are provided to Afrimoney so they can properly call OpenIMIS.

As I said in the README, this needs proper design to include userability, security, and reliability, especially to avoid payment proceeded for the wrong insuree/policy.

Reply:
Could you share that documentation please? We'll have to put that in place, and we need the instructions to do so.